internal/pkg/earpointer/analysis_test.go

```diff
@@ -0,0 +1,512 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package earpointer_test
```

Comment:
It's alright for a test to share `package` with what it's testing.  If it's an intentional decision that you are only interacting with exported members, that's fine.  (The same applies to `state_test.go`, but I missed it on that previous PR.)

We can also do some of these more "clean up" refactor tasks once more of code is in place.

Reply:
I intentionally didn't use "package earpointer" to avoid expose the Test* functions to the user who will use the core analysis. It seems that vet is also using this convention, e.g. https://github.com/golang/tools/blob/master/go/analysis/passes/lostcancel/lostcancel_test.go